### PR TITLE
3 1 5 backport

### PR DESCRIPTION
**Pull request by harsh-ag**
- Updated to 3.2, 3.3, 3.4
- Considering guest users also in users who recenlty purchased report.
- Fixing page_view_events association in product_decorator.
- Not considering returned products in best selling products report.
- Removing undefined scopes in cart_additions, cart_removals, cart_updations and product_views_to_cart_additions_report reports.
- Renaming header months_name to month_name in payment_method_transactions_conversion_rate_report.
- Generalising self.from_union method in query_fragments for n number of queries.
- Using (Arel::SelectManager) in query_fragments according to different version of Spree.
- Fixing returned_products_report by properly joining the tables.
- Considering only completed orders while calculation promotion discount in sales_performance report.
- Considering only completed orders while calculation sales tax in sales_tax report.
- Handling the case when revenue is zero while calculating shipping_cost_percentage in shipping_cost report.
- Considering guest users also in unique_purchases report.
- Fixing user_pool report as it always considered an user as a guest user whether he signs up/in or not.
- Fixing users_not_converted reported as it used to consider users who have initiated an order but didn't complete it.
- Indentation in download.pdf.erb.
- Change in en.yml for the user_pool.

### CODE REVIEW

**Comment by kuche1991** [comment]
Any suggestions if and when this will be merged?

File: app/models/spree/product_decorator.rb [hunk 1]
```diff
@@ -1,3 +1,3 @@
 Spree::Product.class_eval do
-  has_many :page_view_events, -> { viewed.product }, class_name: 'Spree::PageEvent', foreign_key: :target_id
+  has_many :page_view_events, -> { where(activity: 'view') }, class_name: 'Spree::PageEvent', foreign_key: :target_id
```
Comment (tanmay3011) [inline app/models/spree/product_decorator.rb]: @harsh-ag Can you please move `where(activity: 'view')` this to a scope with view as a parameter

File: app/reports/spree/cart_additions_report.rb [hunk 2]
```diff
@@ -19,9 +19,8 @@ def sku
 
     def report_query
       Spree::CartEvent
-        .added
-        .joins(:variant)
-        .joins(:product)
+        .where(activity: 'add')
```
Comment (tanmay3011) [inline app/reports/spree/cart_additions_report.rb]: @harsh-ag  Can you please move this to a scope with 'add' being sent as a parameter

File: app/reports/spree/cart_removals_report.rb [hunk 3]
```diff
@@ -19,9 +19,8 @@ def sku
 
     def report_query
       Spree::CartEvent
-        .removed
-        .joins(:variant)
-        .joins(:product)
+        .where(activity: 'remove')
```
Comment (tanmay3011) [inline app/reports/spree/cart_removals_report.rb]: @harsh-ag :ditto:

File: app/reports/spree/cart_updations_report.rb [hunk 4]
```diff
@@ -22,9 +22,8 @@ def report_query
       quantity_decrease_sql = "CASE WHEN quantity < 0 then spree_cart_events.quantity ELSE 0 END"
 
       Spree::CartEvent
-        .updated
-        .joins(:variant)
-        .joins(:product)
+        .where(activity: 'update')
```
Comment (tanmay3011) [inline app/reports/spree/cart_updations_report.rb]: @harsh-ag  :ditto:

File: app/reports/spree/product_views_to_cart_additions_report.rb [hunk 5]
```diff
@@ -20,9 +20,8 @@ def cart_to_view_ratio
     def report_query
       cart_additions =
         Spree::CartEvent
-          .added
-          .joins(:variant)
-          .joins(:product)
+          .where(activity: 'add')
```
Comment (tanmay3011) [inline app/reports/spree/product_views_to_cart_additions_report.rb]: @harsh-ag :ditto:

**Comment by kuche1991** [comment]
I would really appreciate if this could be merged soon... can I offer any help, to bring this further?

File: app/models/spree/product_decorator.rb (hunk 1 above)
Comment (harsh-ag) [reply app/models/spree/product_decorator.rb]: used predefined scopes of cart_event and page_event in spree_events_tracker gem

**Pull request merged**